src/scripts/main.js

```diff
@@ -1,3 +1,52 @@
 'use strict';
 
-// write code here
+const mainRow = document.querySelector('tr');
+const cells = mainRow.querySelectorAll('th');
+const tableBody = document.querySelector('tbody');
+const tableRow = tableBody.querySelectorAll('tr');
+const rowArray = Array.from(tableRow);
+
+cells.forEach(cell => {
+  cell.addEventListener('click', function() {
```

Comment:
Add event listener to thead (check [event delegation](https://www.freecodecamp.org/news/event-delegation-javascript/)), then you can use `cellIndex` and in your sort function you can one time create `current` and `next` consts and later use those values to sort the table, based on its type. No need to repeate this for each case. 

```js
const thead = document.querySelector('thead');
const tbody = table.querySelector('tbody');
const rows = [...tbody.querySelectorAll('tr')];

thead.addEventListener('click', e => {
  const columnIndex = e.target.cellIndex;

  rows.sort((a, b) => {
    const current = a.cells[cellIndex].innerHTML;
    const next = b.cells[cellIndex].innerHTML;

   // sort based on the type, 
   // for Age and Salary 
   // and fot the rest you can use localCompare

}
```